src/image_cache/mod.rs

```diff
@@ -42,19 +42,20 @@ pub mod errors {
 pub use self::errors::Result;
 use self::errors::*;
 
-pub fn get_image_size_estimate(dimensions: (u32, u32)) -> u32 {
+pub fn get_image_size_estimate(width: u32, height: u32) -> isize {
 	// counting all the mipmaps would add an additionnal multiplier of around ~1.6
 	// but only the gpu textures have mip maps so just multiply by 1.5
-	((dimensions.0 * dimensions.1 * 4) as f32 * 1.5) as u32
+	((width * height * 4) as f32 * 1.5) as isize
 }
 
-pub fn get_anim_size_estimate(frames: &[AnimationFrameTexture]) -> u32 {
-	let mut size = 0;
-	for frame in frames {
-		let dimensions = frame.texture.dimensions();
-		size += get_image_size_estimate(dimensions);
-	}
-	size
+pub fn get_anim_size_estimate(frames: &[AnimationFrameTexture]) -> isize {
+	frames
+		.iter()
+		.map(|frame| {
+			let (width, height) = frame.texture.dimensions();
+			get_image_size_estimate(width, height)
+		})
+		.sum()
```

Comment:
The readibility improvement is questionable here but this is for sure more idiomatic so let's keep it.